desc/geometry/surface.py

```diff
@@ -306,9 +306,54 @@ def _get_transforms(self, grid=None):
         )
         return R_transform, Z_transform
 
-    def compute_curvature(self, params=None, grid=None):
-        """Compute gaussian and mean curvature."""
-        raise NotImplementedError()
+    def _compute_first_fundamental_form(self, R_lmn=None, Z_lmn=None, grid=None):
+        """Compute coefficients for the first fundamental form."""
+        rt = self.compute_coordinates(R_lmn, Z_lmn, grid, dt=1)
+        rz = self.compute_coordinates(R_lmn, Z_lmn, grid, dz=1)
+        E = jnp.sum(rt * rt, axis=-1)
+        F = jnp.sum(rt * rz, axis=-1)
+        G = jnp.sum(rz * rz, axis=-1)
+        return E, F, G
+
+    def _compute_second_fundamental_form(self, R_lmn=None, Z_lmn=None, grid=None):
+        """Compute coefficients for the second fundamental form."""
+        rtt = self.compute_coordinates(R_lmn, Z_lmn, grid, dt=2)
+        rtz = self.compute_coordinates(R_lmn, Z_lmn, grid, dt=1, dz=1)
+        rzz = self.compute_coordinates(R_lmn, Z_lmn, grid, dz=2)
+        n = self.compute_normal(R_lmn, Z_lmn, grid)
+        L = jnp.sum(rtt * n, axis=-1)
+        M = jnp.sum(rtz * n, axis=-1)
+        N = jnp.sum(rzz * n, axis=-1)
+        return L, M, N
+
+    def compute_curvature(self, R_lmn=None, Z_lmn=None, grid=None):
+        """Compute gaussian and mean curvature.
+
+        Parameters
+        ----------
+        R_lmn, Z_lmn: array-like
+            fourier coefficients for R, Z. Defaults to self.R_lmn, self.Z_lmn
+        grid : Grid or array-like
+            toroidal coordinates to compute at. Defaults to self.grid
+            If an integer, assumes that many linearly spaced points in (0,2pi)
+
+        Returns
+        -------
+        K, H, k1, k2 : ndarray, shape(k,)
+            Gaussian, mean and 2 principle curvatures at points specified in grid.
+        """
```

Comment:
This function should be moved to the parent `Surface` class, rather than copying it for both types of surfaces.

Reply:
Agreed, but can make the compute fundamental forms just be not defined in the parent class as they need specific implementation in the subclass